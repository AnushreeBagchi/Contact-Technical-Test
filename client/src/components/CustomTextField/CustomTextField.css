.text-field {
    margin-bottom: 3em !important;
}