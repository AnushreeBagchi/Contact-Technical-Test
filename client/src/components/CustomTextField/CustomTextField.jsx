import React from "react";
import TextField from "@material-ui/core/TextField";
import { connect } from "react-redux";
import { adduser } from "../../store/actions/user";
import "./CustomTextField.css";

class CustomTextField extends React.Component {
  
  render() {
    const onChange = (e, field) => {
      let value = field.type == "number" ? Number(e.target.value) : e.target.value;
      if(field.name == "gender") {
        value = e.target.value.toLowerCase();
      }
      this.props.adduser({ [field.name]: value });
    };

    return (
      <>
      <TextField
        label= {this.props.field.label}
        helperText={this.props.field.helperText}
        minLength={this.props.field.minLength}
        required={this.props.field.required}
        type={this.props.field.type}
        className="text-field"
        onChange={(e) => onChange(e, this.props.field)}
        style={{ margin: 8 }}
        placeholder={this.props.field.defaultValue.toString()}
        fullWidth
        variant="filled"
      ></TextField>
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  state: state,
});
const mapDispatchToProps = (dispatch) => ({
  adduser: (data) => {
    dispatch(adduser(data));
  },
});

export {CustomTextField};

export default connect(mapStateToProps, mapDispatchToProps)(CustomTextField);
