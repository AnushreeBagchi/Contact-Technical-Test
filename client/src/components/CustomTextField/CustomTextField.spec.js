import React from 'react';
import { shallow } from 'enzyme';
import {CustomTextField} from './CustomTextField.jsx';

describe("Tests for CustomTextField ", ()=> {
    const mockField = {
            name: "name",
            label: "name",
            type: "text",
            defaultValue: "Anu",
        };
    it('Renders correctly', () => {
        const wrapper = shallow(<CustomTextField field={mockField} />);
        expect(wrapper).toMatchSnapshot();
      });
})

   
   

