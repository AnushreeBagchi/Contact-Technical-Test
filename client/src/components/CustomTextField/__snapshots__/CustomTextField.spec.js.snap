// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CustomTextField  Renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TextField))
    className="text-field"
    fullWidth={true}
    label="name"
    onChange={[Function]}
    placeholder="Anu"
    style={
      Object {
        "margin": 8,
      }
    }
    type="text"
    variant="filled"
  />
</Fragment>
`;
